pymongo/pool.py

```diff
@@ -886,6 +896,14 @@ def send_cluster_time(self, command, session, client):
         if self.max_wire_version >= 6 and client:
             client._send_cluster_time(command, session)
 
+    def add_server_api(self, command, session):
+        """Add server_api parameters."""
+        if (session and session.in_transaction and
+                not session._starting_transaction):
+            return
+        if self.opts.server_api:
```

Comment:
`self.opts.server_api` is either `None` or an instance of `ServerApi` so I'm not sure I follow.

Reply:
My point here was that since the spec asks us to check if the user passed a value at all, that's exactly what we should do. Since `None` is the default for this kwarg, `if <kwarg name> is not None` is a (more) explicit check for whether it contains its default value. 
It doesn't really affect behavior so if you prefer it the way it is, leave it.

Reply:
If a user passes something that's not a ServerApi they will get an error when creating the client:
```
>>> MongoClient(server_api=False)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pymongo/mongo_client.py", line 678, in __init__
    keyword_opts.cased_key(k), v) for k, v in keyword_opts.items()))
  File "pymongo/mongo_client.py", line 678, in <genexpr>
    keyword_opts.cased_key(k), v) for k, v in keyword_opts.items()))
  File "pymongo/common.py", line 764, in validate
    value = validator(option, value)
  File "pymongo/common.py", line 537, in validate_server_api_or_none
    raise TypeError("%s must be an instance of ServerApi" % (option,))
TypeError: server_api must be an instance of ServerApi
```